### ISSUE TITLE
save labeled multisig wallets in account serialization

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain-wallet-electron#63]
We only need to store a mapping between addresses and labels, more info is not required, as all other info is received from the explorer node(s).

**Comment by GlenDC** [comment]
After a discussion we'll do it as follows:

The serialized account will have in its payload next to the regular wallets property (`wallets`) also a property `multisig_wallets` which will also be an array of which each element has the following layout:

```javascript
{
     "address": "03...",
     "label": "label of wallet", // will always be non-empty, otherwise we don't store it
}
```

The TFChain API will provide a new class for the multisig. wallets:

```python
class MultiSignatureWallet:
    # relevant for API user

    # methods
    def transaction_new(self):
        """
        :returns: a transaction builder that allows for the building of transactions
        """
        pass

    def transaction_sign(self, transaction):
        """
        :returns: signs an existing transaction
        """
        pass

    # properties
    @property
    def wallet_label(self):
        """
        :returns: the label (str) of the wallet if defined, None otherwise
        """
        pass

    @property
    def address(self):
        """
        :returns: the multisig address of the wallet
        """
        pass

    @property
    def balance(self);
        """
        :returns: the balance object async as a promise
        """
        pass

    # other methods
    def clone(self):
        pass
```

The `Account` object from the public API will also have the following new properties and methods:

```python
def labeled_multisig_wallets(self):
    """
    :returns: all labeled multisig wallets linked to this wallet
    """

def multisig_wallet_label_get(self, address):
    """
    :returns: None if no label is attached to address or a label if it is
    """

def multisig_wallet_label_update(self, address, label):
    """
    Delete (label=None||"") or update the label for an address.
    """
```

The above methods use and populate the data that we serialized as part of the serialized account as proposed in the beginning of this comment.

An `MultiSignatureWallet` can be created using a new method in the `Account` object:

```python
def multisig_wallet_new(balance):
    """
    Creates a new multisig wallet for the given
    MultiSignatureBalance object.
    """
```

> (!) The `MultiSignatureWallet` will have embedded the wallets owned by the account and that own the multisig wallet.
> (!!) NOTE: this is not an ideal design, see if we cannot make it possible to create this without requiring to fetch a balance first. Or perhaps this is OK, need to see.

**Comment by GlenDC** [comment]
Closed by 67592ea66058a9b5d9eea1114ffd175dc4304af6.
